cpp/src/arrow/compute/kernels/scalar_if_else.cc

```diff
@@ -770,6 +770,144 @@ struct IfElseFunctor<Type, enable_if_base_binary<Type>> {
   }
 };
 
+template <typename Type>
+struct IfElseFunctor<Type, enable_if_fixed_size_binary<Type>> {
+  // A - Array, S - Scalar, X = Array/Scalar
+
+  // SXX
+  static Status Call(KernelContext* ctx, const BooleanScalar& cond, const Datum& left,
+                     const Datum& right, Datum* out) {
+    auto byte_width =
+        std::static_pointer_cast<FixedSizeBinaryType>(left.type())->byte_width();
```

Comment:
maybe `checked_cast<const FixedSizeBinaryType&>(*left.type())` or `checked_pointer_cast<FixedSizeBinaryType>(left.type())` (will be dynamic cast in debug, static cast in release)